src/main/groovy/org/jetbrains/intellij/IntelliJPlugin.groovy

```diff
@@ -312,44 +319,92 @@ class IntelliJPlugin implements Plugin<Project> {
+            task.args(["traverseUI", "$project.buildDir/$SEARCHABLE_OPTIONS_DIR_NAME", "true"])
+            task.dependsOn(PREPARE_SANDBOX_TASK_NAME)
+            task.onlyIf {
+                def number = Utils.ideaBuildNumber(Utils.ideaSdkDirectory(extension))
+                VersionNumber.parse(number[number.indexOf('-') + 1..-1]) >= VersionNumber.parse("191.2752")
+            }
+        }
+    }
+
+    private static void prepareConventionMappingsForRunIdeTask(@NotNull Project project, @NotNull IntelliJPluginExtension extension, 
+                                                               @NotNull RunIdeTask task) {
+        def prepareSandboxTask = project.tasks.findByName(PREPARE_SANDBOX_TASK_NAME) as PrepareSandboxTask
+        task.conventionMapping.map("ideaDirectory", { Utils.ideaSdkDirectory(extension) })
+        task.conventionMapping.map("requiredPluginIds", { Utils.getPluginIds(project) })
+        task.conventionMapping.map("configDirectory", { prepareSandboxTask.getConfigDirectory() })
+        task.conventionMapping.map("pluginsDirectory", { prepareSandboxTask.getDestinationDir() })
+        task.conventionMapping.map("systemDirectory", {
+            project.file(Utils.systemDir(extension.sandboxDirectory, false))
+        })
+        task.conventionMapping("executable", {
+            def jbreResolver = new JbreResolver(project)
+            def jbreVersion = task.getJbreVersion()
+            if (jbreVersion != null) {
+                def jbre = jbreResolver.resolve(jbreVersion)
+                if (jbre != null) {
+                    return jbre.javaExecutable
                 }
-                def builtinJbreVersion = Utils.getBuiltinJbreVersion(Utils.ideaSdkDirectory(extension))
-                if (builtinJbreVersion != null) {
-                    def builtinJbre = jbreResolver.resolve(builtinJbreVersion)
-                    if (builtinJbre != null) {
-                        return builtinJbre.javaExecutable
-                    }
-                    LOG.warn("Cannot resolve builtin JBRE $builtinJbreVersion. Falling local Java.")
+                LOG.warn("Cannot resolve JBRE $jbreVersion. Falling back to builtin JBRE.")
+            }
+            def builtinJbreVersion = Utils.getBuiltinJbreVersion(Utils.ideaSdkDirectory(extension))
+            if (builtinJbreVersion != null) {
+                def builtinJbre = jbreResolver.resolve(builtinJbreVersion)
+                if (builtinJbre != null) {
+                    return builtinJbre.javaExecutable
                 }
-                return Jvm.current().javaExecutable.absolutePath
-            })
+                LOG.warn("Cannot resolve builtin JBRE $builtinJbreVersion. Falling local Java.")
+            }
+            return Jvm.current().javaExecutable.absolutePath
+        })
+    }
 
-            task.dependsOn(PREPARE_SANDBOX_TASK_NAME)
+    private static void configureJarSearchableOptions(@NotNull Project project) {
```

Comment:
I think it make sense to extract it into a separate class to make `IntellijPlugin` class smaller